Assertion error while trying to create ROM dump for xaga 14.0.3TLOCNXM

I encounter this error while running ./dumpyara.sh <file>

`Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/db-rom/dumpyara/.venv/lib/python3.9/site-packages/twrpdtgen/__main__.py", line 11, in <module>
    main()
  File "/home/db-rom/dumpyara/.venv/lib/python3.9/site-packages/twrpdtgen/main.py", line 37, in main
    device_tree = DeviceTree(image=args.image)
  File "/home/db-rom/dumpyara/.venv/lib/python3.9/site-packages/twrpdtgen/device_tree.py", line 66, in __init__
    self.device_info = DeviceInfo(self.build_prop)
  File "/home/db-rom/dumpyara/.venv/lib/python3.9/site-packages/sebaubuntu_libs/libandroid/device_info.py", line 117, in __init__
    self.codename = self.get_first_prop(DEVICE_CODENAME)
  File "/home/db-rom/dumpyara/.venv/lib/python3.9/site-packages/sebaubuntu_libs/libandroid/device_info.py", line 177, in get_first_prop
    raise AssertionError(f'Property {props[0]} could not be found in build.prop')
AssertionError: Property ro.product.bootimage.device could not be found in build.prop`

Is there any way to overcome this other than to manually change the files?

--- a/dumpyara.sh
+++ b/dumpyara.sh
@@ -225,17 +225,14 @@ manufacturer=$(echo "$manufacturer" | tr '[:upper:]' '[:lower:]' | tr -dc '[:pri
 printf "# %s\n- manufacturer: %s\n- platform: %s\n- codename: %s\n- flavor: %s\n- release: %s\n- id: %s\n- incremental: %s\n- tags: %s\n- fingerprint: %s\n- is_ab: %s\n- brand: %s\n- branch: %s\n- repo: %s\n" "$description" "$manufacturer" "$platform" "$codename" "$flavor" "$release" "$id" "$incremental" "$tags" "$fingerprint" "$is_ab" "$brand" "$branch" "$repo" > "$PROJECT_DIR"/working/"${UNZIP_DIR}"/README.md
 cat "$PROJECT_DIR"/working/"${UNZIP_DIR}"/README.md
 
-# create TWRP device tree if possible
-if [[ "$is_ab" = true ]]; then
-    twrpimg="$PROJECT_DIR"/working/"${UNZIP_DIR}"/"boot.img"
-else
-    twrpimg="$PROJECT_DIR"/working/"${UNZIP_DIR}"/"recovery.img"
-fi
-if [[ -f "${twrpimg}" ]]; then
-    twrpdt="$PROJECT_DIR"/working/"${UNZIP_DIR}"/twrp-device-tree
-    python3 -m twrpdtgen "$twrpimg" --output "$twrpdt"
-    if [[ "$?" = 0 ]]; then
-        [[ ! -e "$twrpdt"/README.md ]] && curl https://raw.githubusercontent.com/wiki/SebaUbuntu/TWRP-device-tree-generator/4.-Build-TWRP-from-source.md > "$twrpdt"/README.md
+# Generate AOSP device tree
+if python3 -c "import aospdtgen"; then
+    echo "aospdtgen installed, generating device tree"
+    mkdir -p "${PROJECT_DIR}/working/${UNZIP_DIR}/aosp-device-tree"
+    if python3 -m aospdtgen . --output "${PROJECT_DIR}/working/${UNZIP_DIR}/aosp-device-tree"; then
+        echo "AOSP device tree successfully generated"
+    else
+        echo "Failed to generate AOSP device tree"
     fi
 fi
 

--- a/setup.sh
+++ b/setup.sh
@@ -27,4 +27,4 @@ fi
 # Create virtual environment and install packages
 python3 -m venv .venv
 source .venv/bin/activate
-"$PIP" install backports.lzma extract-dtb protobuf pycrypto docopt zstandard twrpdtgen
+"$PIP" install aospdtgen backports.lzma extract-dtb protobuf pycrypto docopt zstandard

